includes/tracks/events/class-wc-admin-setup-wizard-tracking.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * WooCommerce Admin Setup Wizard Tracking
+ *
+ * @package WooCommerce\Tracks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds actions to track usage of the WooCommerce Onboarding Wizard.
+ */
+class WC_Admin_Setup_Wizard_Tracking {
+	/**
+	 * Init tracking.
+	 */
+	public static function init() {
+		if ( empty( $_GET['page'] ) || 'wc-setup' !== $_GET['page'] ) { // WPCS: CSRF ok, input var ok.
+			return;
+		}
+
+		// phpcs:disable WordPress.Security.NonceVerification.NoNonceVerification
+		$current_step = isset( $_GET['step'] ) ? sanitize_key( $_GET['step'] ) : '';
+		if ( ! empty( $_POST['save_step'] ) ) {
```

Comment:
maybe more readable if you return early?
```
if ( empty( $_POST['save_step'] ) ) {
    return;
}
switch ( $current_step ) { ...
```

Reply:
Sure, I think we can do that.  We have other hooks in other PRs that don't fall under this condition, but I think we can re-order them to get them working again.

Reply:
If it reads better this way, feel free to leave it

Reply:
Extracted the save events into a separate method in https://github.com/woocommerce/woocommerce/pull/22860/commits/194be16dc6c1e69798026fdbd85fdfd918d0e906 to clean this up a bit and make it more readable.